instaclone/src/Authentication/Authenticate.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import myStorage from '../Storage'
+import Login from '../Components/Login/Login'
+
+const Authenticate = App => class extends React.Component{
+    constructor(props){
+        super(props);
+        this.state= {
+            loggedIn: false,
+            user: ""
+        }
+    }
+
+    //  this checks to see if there is a user in local storage
+    componentDidMount(){
+        if(myStorage.getObject('user')){
+            this.setState({loggedIn: true})
+            this.setState({user:myStorage.getObject('user')})
+        }
+        
+    }
+
+     // updates local storage with the username and also sets the state of apps this.state.user
+    login = (username, password) => { 
+        myStorage.setObject('user', {username:username});
+        this.setState({user:  myStorage.getObject('user').username})
```

Comment:
This is great! However, how might you refactor so that instead of just storing a username, it checks the username against an object of users and passwords to see if the user exists and if the password is correct and then logs the user in if it is?